File: Core/YABACore/src/commonMain/kotlin/dev/subfly/yabacore/model/annotation/EpubAnnotationExtras.kt
package dev.subfly.yabacore.model.annotation

import kotlinx.serialization.Serializable

/** Serialized into [dev.subfly.yabacore.database.entities.AnnotationEntity.extrasJson] for [AnnotationType.EPUB]. */
@Serializable
data class EpubAnnotationExtras(
    /** epub.js CFI range for the highlight (inclusive). */
    val cfiRange: String,
)

File: Darwin/YABACore/Model/Annotation/EpubAnnotationExtras.swift
//
//  EpubAnnotationExtras.swift
//  YABACore
//
//  Parity with Compose `EpubAnnotationExtras`.
//

import Foundation

public struct EpubAnnotationExtras: Sendable, Codable, Equatable {
    /// epub.js CFI range for the highlight (inclusive).
    public var cfiRange: String

    public init(cfiRange: String) {
        self.cfiRange = cfiRange
    }
}
